datadog/resource_datadog_dashboard.go

```diff
@@ -4901,8 +4932,10 @@ func buildDatadogEventQuery(data map[string]interface{}) datadogV1.FormulaAndFun
 	eventQuery.SetIndexes(indexes)
 
 	if terraformSearches, ok := data["search"].([]interface{}); ok && len(terraformSearches) > 0 {
-		terraformSearch := terraformSearches[0].(map[string]interface{})
-		eventQuery.Search = datadogV1.NewFormulaAndFunctionEventQueryDefinitionSearch(terraformSearch["query"].(string))
+		terraformSearch, ok := terraformSearches[0].(map[string]interface{})
+		if ok {
+			eventQuery.Search = datadogV1.NewFormulaAndFunctionEventQueryDefinitionSearch(terraformSearch["query"].(string))
+		}
```

Comment:
This prevents a crash when `search.query = ""`, but it raises the same issue mentioned [here](https://github.com/DataDog/terraform-provider-datadog/pull/951/files#r582418101).

Reply:
Resolved this by using a newly build string validator in [this commit](https://github.com/DataDog/terraform-provider-datadog/pull/953/commits/4a225543834ef2aa85a9579eba380986c580a915).